op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,255 @@
+	if h.server != nil {
+		h.log.Info("closing WebSocket server")
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		err := h.server.Shutdown(ctx)
+		if err != nil {
+			h.log.Error("Error shutting down WebSocket server", "err", err)
+		}
+		h.log.Info("WebSocket server closed")
+	}
+}
+
+// BroadcastMessage sends a message to all connected WebSocket clients
+func (h *Handler) BroadcastMessage(message []byte) {
+	h.hub.broadcast <- message
+}
+
+func (h *Handler) startWebSocketServer(_ context.Context) error {
+	if h.cfg.WebsocketServerPort <= 0 {
+		return fmt.Errorf("WebSocket server port not configured or invalid: %d", h.cfg.WebsocketServerPort)
+	}
+
+	h.hub = newHub()
+	go h.hub.run()
+
+	// Create HTTP server with WebSocket endpoint
+	mux := http.NewServeMux()
+	mux.HandleFunc("/ws", h.handleWebSocket)
+
+	// Start HTTP server
+	h.server = &http.Server{
+		Addr:    fmt.Sprintf(":%d", h.cfg.WebsocketServerPort),
+		Handler: mux,
+	}
+
+	h.log.Info("starting WebSocket server", "port", h.cfg.WebsocketServerPort)
+	go func() {
+		if err := h.server.ListenAndServe(); err != nil && !errors.Is(err, http.ErrServerClosed) {
+			h.log.Error("WebSocket server error", "err", err)
+		}
+	}()
+
+	return nil
+}
+
+// handleWebSocket handles WebSocket upgrade requests
+func (h *Handler) handleWebSocket(w http.ResponseWriter, r *http.Request) {
+	h.serveWs(w, r)
+}
+
+// listenToRollupBoost listens for messages from the rollup boost WebSocket
+func (h *Handler) listenToRollupBoost(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			// Try to connect if not connected indefinitely
+			if h.rollupBoostConn == nil {
+				h.log.Info("reconnecting to rollup boost WebSocket", "url", h.cfg.RollupBoostWsURL)
+
+				// Connect with timeout
+				dialCtx, _ := context.WithTimeout(ctx, 5*time.Second)
+				conn, _, err := websocket.Dial(dialCtx, h.cfg.RollupBoostWsURL, nil)
+
+				if err != nil {
+					h.log.Warn("failed to connect to rollup boost WebSocket, will retry",
+						"err", err, "retryIn", reconnectDelay)
+					// add a metric for the number of times we've tried to connect
+					h.metrics.RecordRollupBoostConnectionAttempts(false)
+					time.Sleep(reconnectDelay)
+					continue
+				}
+
+				h.rollupBoostConn = conn
+				h.log.Info("successfully connected to rollup boost WebSocket")
+			}
+
+			// Read with timeout
+			readCtx, _ := context.WithTimeout(ctx, 30*time.Second)
+			messageType, message, err := h.rollupBoostConn.Read(readCtx)
+
+			if err != nil {
+				h.log.Warn("error reading from rollup boost WebSocket", "err", err)
+				// Close and reset for reconnection
+				h.rollupBoostConn.Close(websocket.StatusInternalError, "read error")
+				h.rollupBoostConn = nil
+				continue
+			}
+
+			// Only process valid message types
+			if messageType == websocket.MessageText || messageType == websocket.MessageBinary {
+				h.handleRollupBoostMessage(ctx, message)
+			}
```

Comment:
```suggestion
			h.handleRollupBoostMessage(ctx, message)
```

Same reasoning as this comment - https://github.com/ethereum-optimism/optimism/pull/15681#discussion_r2128136029